pulser-core/pulser/errors/__init__.py

```diff
@@ -0,0 +1,301 @@
+"""Errors raised by Pulser.
+
+Note: As of this writing, the specific error strings for these classes are
+maintained for backwards compatibility. However, they may change in a future
+version, so if you need to catch a specific error, you should rather rely
+upon its class and fields.
+"""
+
+# Avoid circular dependencies due to type hints, part 1.
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    # Avoid circular dependencies due to type hints, part 2.
+    from pulser.devices._device_datacls import BaseDevice
+    from pulser.register.base_register import QubitId
+    from pulser.register.base_register import RegisterLayout
+
+
+class PulserError(ValueError):
```

Comment:
I'm not sure all `PulserError`s should be `ValueError`s of some kind. Perhaps it would be better to call it `PulserValueError`?

Reply:
Not entirely sure I follow.

As far as I understand, we agree that all the errors raised by Pulser (at this stage, at least) should remain instances of `ValueError`, for the backwards compatibility. Do you just want me to rename this class to leave the name `PulserError` open for future errors? Happy to do that.

Reply:
Although all the errors you are replacing here are `ValueErrors`, not all the errors we raise in Pulser are `ValueError`s, so I think it would be misleading to have `PulseError` error be a `ValueError` sublclass. Renaming sounds good to me

Reply:
> We already have custom exceptions to handle the serialization/deserialization errors (see https://github.com/pasqal-io/Pulser/blob/develop/pulser-core/pulser/json/exceptions.py), would it make sense to move these in this file ? These custom exceptions are ValueErrors, I agree with Henrique that it would be better to rename this class PulserValueError or something like of the kind :)

Can we maybe handle (de)serialization in a followup? I'd like to keep the size of this PR reasonable.